src/components/card/Card.js

```diff
@@ -0,0 +1,65 @@
+import "./card.css";
+import moment from "moment";
+import { useEffect, useState } from "react";
+import LogoImg from "../../assets/common/img_logo.png";
+
+const Card = ({ cardData }) => {
+  const [ago, setAgo] = useState(0);
+  const [createdAtFormat, setCreatedAtFormat] = useState(cardData.createdAt);
+
+  const calculateTimeAgo = (createdAt) => {
+    const createdDate = moment(createdAt, "YYYY-MM-DDTHH:mm:ss[Z]");
+    const currentDate = moment();
+    const diff = currentDate.diff(createdDate, "seconds");
+
+    if (diff < 120) {
+      return "1 minute ago";
+    } else if (diff <= 3540) {
+      return `${Math.floor(diff / 60)} minutes ago`;
+    } else if (diff < 3600) {
+      return "1 hour ago";
+    } else if (diff <= 82800) {
+      return `${Math.floor(diff / 3600)} hours ago`;
+    } else if (diff < 86400) {
+      return "1 day ago";
+    } else if (diff <= 2592000) {
+      return `${Math.floor(diff / 86400)} days ago`;
+    } else if (diff <= 28512000) {
+      return `${Math.floor(diff / 2592000)} months ago`;
+    } else if (diff <= 31536000) {
+      return "1 year ago";
+    } else {
+      return `${Math.floor(diff / 31536000)} years ago`;
+    }
+  };
+
+  const openUrl = () => {
+    window.open(cardData.url, "__blank");
+  };
+
+  useEffect(() => {
+    setAgo(calculateTimeAgo(cardData.createdAt));
+    setCreatedAtFormat(moment(cardData.createdAt).format("YYYY.MM.DD"));
+  }, [cardData]);
```

Comment:
cardData는 props라 값이 변할때 렌더링이 다시 이루어지기 때문에 ago, createdAtFormat 값을 상태로 관리하지 않아도 될 것 같은 생각이 듭니다. 혹시 특별한 이유가 있나요??

Reply:
ago는 createdAt과 현재 시간의 차이, createdAtFormat은 createdAt의 형태를 yyyy.mm.dd 형태 변형한 값들인데 cardData값이 넘어오거나 업데이트가 되면 createdAt을 기준으로 해당 값들을 다시 설정해주고 관리해야 한다고 생각해서 이렇게 작성했습니다. 
혹시 ago, createdAtFormat 값을 상태로 관리하지 않아도 된다면 어떤식으로 처리해주는 것이 좋은가요? 혹시 아래와 같은 코드를 말씀하시는걸까요?

```suggestion
const ago = calculateTimeAgo(cardData.createdAt)
const createdAtFormat = cardData.createdAt.format('YYYY.MM.DD'))
```

Reply:
네네 맞습니다. cardData가 useState를 이용하여 선언되었기 때문에 이 값이 제대로 mutate 된다면 cardData의 데이터의 참조가 바뀔 것임을 보장하기 때문에 useEffect 없이도 ago, createdAtFormat이 잘 반영되어서 업데이트 될거에요.

Reply:
https://react.dev/learn/updating-arrays-in-state#making-other-changes-to-an-array

Reply:
아아 이해했습니다. 말씀주신 방향으로 수정 진행하겠습니다.👍